src/main/java/com/parkhomovsky/bookstore/service/implementation/CartItemServiceImpl.java

```diff
@@ -0,0 +1,129 @@
+package com.parkhomovsky.bookstore.service.implementation;
+
+import com.parkhomovsky.bookstore.dto.cart.ShoppingCartDto;
+import com.parkhomovsky.bookstore.dto.item.AddCartItemRequestDto;
+import com.parkhomovsky.bookstore.dto.item.CartItemDto;
+import com.parkhomovsky.bookstore.dto.item.CreateCartItemRequestDto;
+import com.parkhomovsky.bookstore.exception.EntityNotFoundException;
+import com.parkhomovsky.bookstore.exception.InvalidRequestParametersException;
+import com.parkhomovsky.bookstore.exception.UserNotAuthenticatedException;
+import com.parkhomovsky.bookstore.mapper.CartItemMapper;
+import com.parkhomovsky.bookstore.mapper.ShoppingCartMapper;
+import com.parkhomovsky.bookstore.model.Book;
+import com.parkhomovsky.bookstore.model.CartItem;
+import com.parkhomovsky.bookstore.model.ShoppingCart;
+import com.parkhomovsky.bookstore.repository.book.BookRepository;
+import com.parkhomovsky.bookstore.repository.item.CartItemRepository;
+import com.parkhomovsky.bookstore.service.CartItemService;
+import com.parkhomovsky.bookstore.service.ShoppingCartService;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import org.springframework.validation.annotation.Validated;
+
+@Service
+@RequiredArgsConstructor
+@Validated
+public class CartItemServiceImpl implements CartItemService {
```

Comment:
Please review this class again and repeat SOLID principles, especially Single Responsibility